src/modules/CourseSearch/components/RecommendationText/index.tsx

```diff
@@ -0,0 +1,100 @@
+import { useLazyQuery } from '@apollo/client'
+import styled from '@emotion/styled'
+import { Link, Typography } from '@mui/material'
+import useGoogleOptimize from '@react-hook/google-optimize'
+import { observer } from 'mobx-react'
+import { useEffect, useState } from 'react'
+
+import { Analytics } from '@/common/context/Analytics/components/Analytics'
+import { useCourseGroup } from '@/common/hooks/useCourseGroup'
+import { collectLogEvent } from '@/services/logging'
+import { courseCartStore } from '@/store'
+import { RecommendationParam, RecommendationResponse, RECOMMENDATION_QUERY } from '@/services/apollo/query/recommendation'
+import { useTheme } from '@emotion/react'
+import { useCourseSearchProvider } from '@/modules/CourseSearch/context/CourseSearch/hooks/useCourseSearchProvider'
+import { SearchCourseVars } from '@/services/apollo/query/searchCourse'
+import { useSearchCourseQueryParams } from '@/modules/CourseSearch/hooks/useSearchCourseQueryParams'
+
+const RecommendationItem = styled(Link)`
+  color: ${({theme}) => theme.palette.highlight.indigo[700]};
+  margin-left: 1em;
+`
+
+const RecommendationText: React.FC<{variant: string}> = observer((props: { variant: string })  => {
+  const variant = props.variant
+  const courseGroup = useCourseGroup()
+  const selectedCourses =courseCartStore.shopItems.map((item) => ({
+    courseNo: item.courseNo,
+    semesterKey: {
+      semester: item.semester,
+      studyProgram: item.studyProgram,
+      academicYear: item.academicYear,
+    },
+  }))
+  const {courseSearchQuery} = useCourseSearchProvider()
+  const {setFilter} = useSearchCourseQueryParams()
+  const [lastSearchQuery, setLastSearchQuery] = useState<SearchCourseVars | undefined>(undefined)
+  const [fetchRecommendation, { data }] = useLazyQuery<RecommendationResponse, RecommendationParam>(RECOMMENDATION_QUERY)
+
+  const visibleRecommendation = data && data.recommend.courses.length > 0 ? data.recommend.courses.slice(0, 6) : null
+
+  useEffect(() => {
+    const visibleRecommendation = data && data.recommend.courses.length > 0 ? data.recommend.courses.slice(0, 6) : null
+    collectLogEvent({
+      kind: 'track',
+      message: 'displayed recommendation',
+      additionalData: {
+        variant,
+        display: JSON.stringify(visibleRecommendation),
+      },
+    })
+  }, [data, variant])
+
+  useEffect(() => {
+    if (JSON.stringify(courseSearchQuery.variables) === JSON.stringify(lastSearchQuery)) 
```

Comment:
Why do we fetch when search query changes?

Reply:
So that we can perform layout change (due to differing length of recommendation fetched) without user noticing. This fix what @new5558 describe above.